controllers/apps/apimanager_controller.go

```diff
@@ -362,13 +362,20 @@ func (r *APIManagerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return nil
 	}
 
+	watchedByDataOnlySecretLabelPredicate := helper.GetWatchedByDataOnlySecretLabelPredicate("system")
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&appsv1alpha1.APIManager{}).
 		Watches(
 			&v1.Secret{},
 			handler.EnqueueRequestsFromMapFunc(secretToApimanagerEventMapper.Map),
 			builder.WithPredicates(labelSelectorPredicate),
 		).
+		Watches(
+			&v1.Secret{},
+			handler.EnqueueRequestsFromMapFunc(secretToApimanagerEventMapper.Map),
+			builder.WithPredicates(watchedByDataOnlySecretLabelPredicate),
+		).
```

Comment:
@valerymo I think it would be best if we remove the partial watched-by improvements from this PR and do it entirely in [THREESCALE-11395](https://issues.redhat.com/browse/THREESCALE-11395).

We also want to use the same implementation (tracking changes to secret's `.data`  using the master hashed secret) from the [apicast-operator PR](https://github.com/3scale/apicast-operator/pull/233).

Reply:
Hi @carlkyrillos  I will remove my Predicate based watching from the PR from here to make Jiras purpose clear.   I will send update when done.  
Also - 
I added suggestion to [THREESCALE-11395](https://issues.redhat.com/browse/THREESCALE-11395)  to consider combined 
 approach - use both methods. 
Thank you for comments!

Reply:
Done. Thank you